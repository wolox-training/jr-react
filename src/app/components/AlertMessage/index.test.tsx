import React from 'react';
import { render, screen } from '@testing-library/react';

import AlertComponent from './index';

describe('Component Alert', () => {
  it('Renders successfully', () => {
    const { asFragment } = render(<AlertComponent message="Registro exitoso" type="success" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Display the alert on the screen', () => {
    render(<AlertComponent message="Registro exitoso" type="success" />);
    expect(screen.getByRole('alert')).toHaveTextContent('Registro exitoso');
  });

  it('the appearance of the alert changes depending on the type that is defined', () => {
    render(<AlertComponent message="Registro exitoso" type="success" />);
    expect(screen.getByRole('alert')).toHaveClass('success');
  });
});
