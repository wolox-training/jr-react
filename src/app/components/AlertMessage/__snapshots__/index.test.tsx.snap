// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Alert Renders successfully 1`] = `
<DocumentFragment>
  <div
    class="alertMessage success"
    data-testid="alert-success"
    role="alert"
  >
    <h2
      class="title"
    >
      AlertMessage:titleSuccess
    </h2>
    Registro exitoso
  </div>
</DocumentFragment>
`;
